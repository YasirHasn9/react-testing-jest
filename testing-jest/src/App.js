import React from 'react';
import './App.css';

function App() {
  return (
    <div className="App">
    <h1>testing the snapshot/jest</h1>
    </div>
  );
}

export default App;
