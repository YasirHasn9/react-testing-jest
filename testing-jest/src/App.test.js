import React from "react";
import * as rtl from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import App from "./App";

test("App" , () => {
  const wrapper =  rtl.render(<App />)
  const e = wrapper.getByText(/testing the snapshot\/jest/i)
  expect(e).toBeVisible()

})

test("snapshot test" , () => {
  const wrapper = rtl.render(<App />)
  expect(wrapper.asFragment()).toMatchSnapshot()
})
