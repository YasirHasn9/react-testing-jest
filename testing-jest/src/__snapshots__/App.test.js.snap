// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      testing the snapshot/jest
    </h1>
  </div>
</DocumentFragment>
`;
